pkg/api/routes.go

```diff
@@ -552,6 +549,35 @@ func SetupRoutes(node *masa.OracleNode) *gin.Engine {
 		})
 	})
 
-	router.GET("/swagger/*any", ginSwagger.WrapHandler(swaggerFiles.Handler, ginSwagger.DefaultModelsExpandDepth(-1)))
 	return router
 }
+
+func setupSwaggerHandler(router *gin.Engine) {
+	// Get the current file's directory
+	_, currentFile, _, _ := runtime.Caller(0)
+	currentDir := filepath.Dir(currentFile)
+
+	// Construct the path to the swagger.html file
+	swaggerTemplate := filepath.Join(currentDir, "templates", "swagger.html")
```

Comment:
I'm confused, why we need to bring the swagger.html default template here and call it with a custom handler?

Reply:
swagger doesn't automatically set HTTP/HTTPS protocol for the generated URLs. This allows that to happen

Reply:
> swagger doesn't automatically set HTTP/HTTPS protocol for the generated URLs. This allows that to happen

IIIRC it is enough to not setup a URL, which you did as part of this PR - then swagger will follow up and show links within the same relative URL - or are we setting up HTTPs manually here from the http server without a proxy on front? are we doing mTLS?

I'm guessing this is a limitation of using `gin`, as with other frameworks this isn't an issue.

> Since it's public facing, this allows us to customize the page further if we need to.

Understand, but I would keep things very strict to the problem that are trying to solve and avoid to add unused code.

Reply:
correct, it's not enough. Without modifying this way, it autoselects https or http for that dropdown selector based purely on which one is listed first, and it uses that when generating the urls. Why it doesn't do the protocol relatively, I do not know. But, my function just performs the same work that is done if a user were to switch http/https to the correct setting manually after detecting the protocol used to access the page. Setting the base url to "" works in terms of getting rid of the domain and port and using the one the page was accessed with, but it doesn't set the protocol.

Reply:
While this is not a huge deal (having to select the protocol manually if you are not using whichever one was listed first in the config). We have other templates in this area (pkg/api/templates/, and swagger was designed to be overridden this way.

Is there a better or simpler way to do this? almost certainly yes... but I need to move on to other things and this works fine.